src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -323,12 +360,37 @@ public void init(ImportContext context) throws IOException {
     public void init(ImportContext context, boolean prepData) throws IOException {
         context.reattach(catalog);
 
-        ImportData data = context.getData();
-        if (data != null) {
-            addTasks(context, data, prepData); 
+        try {
+            ImportData data = context.getData();
+            if (data != null) {
+                if (data instanceof RemoteData) {
+
+                    data = ((RemoteData) data).resolve(this);
+                    context.setData(data);
+                }
+
+                addTasks(context, data, prepData);
+            }
+
+            // switch from init to pending as needed
+            if (context.getState() == ImportContext.State.INIT) {
+                context.setState(ImportContext.State.PENDING);
+            }
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Failed to resolve remote data", e);
```

Comment:
If we are doing a REST import (for example), is there any way of viewing capturing this error? As it stands, it looks like there is no way (without looking at the GeoServer logs) of telling the difference between an empty remote directory and an error - both result in a completed import with no tasks/layers.
Maybe an error field could be included in the import context, similar to tasks?